### PR TITLE
[wip] Doc cleanup for examples/color/color_cycle and examples/animation/animated_histogram

### PR DESCRIPTION
**Pull request by johnh2o2**
## PR Summary

Scipy 2017 sprint! Working on improving the documentation with better
sphinx-gallery integration. I've added some more documentation to
the `examples/animation/animated_histogram` and the `examples/color/color_cycle` sections. I've also moved the `color_cycle` from `examples/color` to `tutorials/02_Intermediate`, since it seemed more like something users would be interested in doing regularly, and I've worded it more like a tutorial. Not sure how people feel about that.

## PR Checklist

- [n/a] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [n/a] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [n/a] Added an entry to doc/users/whats_new.rst if major new feature
- [n/a] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by johnh2o2** [comment]
This also addresses the doc issues in #8885

**Comment by choldgraf** [comment]
Nice! I'll give it a look when I'm back in Berkeley and will give some comments!

(if you're looking for something to work on quickly, I noticed that you use single tickmarks for comments, while rST uses double backticks (``). Also make sure that you've always got whitespace around operators (e.g.,`a = 2 + 2`, not `a = 2+2`)

File: examples/animation/histogram.py [hunk 1]
```diff
@@ -30,12 +28,20 @@
...
 
-# here comes the tricky part -- we have to set up the vertex and path
-# codes arrays using moveto, lineto and closepoly
-
-# for each rect: 1 for the MOVETO, 3 for the LINETO, 1 for the
-# CLOSEPOLY; the vert for the closepoly is ignored but we still need
-# it to keep the codes aligned with the vertices
+###############################################################################
+# Here comes the tricky part -- we have to set up the vertex and path codes
+# arrays using `plt.Path.MOVETO`, `plt.Path.LINETO` and `plt.Path.CLOSEPOLY`
```
Comment (choldgraf) [inline examples/animation/histogram.py]: `` double backticks for rST (do this here and in all subsequent uses of backticks)

File: examples/animation/histogram.py [hunk 2]
```diff
@@ -30,12 +28,20 @@
...
+# for each rect.
+#
+# * We need 1 `MOVETO` per rectangle, which sets the initial point.
+# * We need 3 `LINETO`'s, which tell `matplotlib` to draw lines from vertex 1
+#   to vertex 2, v2 to v3, and v3 to v4.
+# * We then need one `CLOSEPOLY` which tells matplotlib to draw a line from the
+#   v4 to our initial vertex (the `MOVETO` vertex), in order to close the
+#   polygon.
+#
+# Note: the vertex for `CLOSEPOLY` is ignored, but we still need a placeholder
```
Comment (choldgraf) [inline examples/animation/histogram.py]: this could be `.. note::`

File: tutorials/02_intermediate/color_cycle.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+"""
+from cycler import cycler
+import numpy as np
+import matplotlib.pyplot as plt
+
+###############################################################################
+# First we'll generate some sample data, in this case, four offset sine
+# curves.
+x = np.linspace(0, 2 * np.pi, 50)
+offsets = np.linspace(0, 2*np.pi, 4, endpoint=False)
```
Comment (choldgraf) [inline tutorials/02_intermediate/color_cycle.py]: spaces around the *

File: tutorials/02_intermediate/color_cycle.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+yy = np.transpose([np.sin(x + phi) for phi in offsets])
+
+###############################################################################
+# Now `yy` has shape
+print(yy.shape)
+
+###############################################################################
+# So `yy[:,i]` will give you the `i`-th offset sine curve. Let's set the
```
Comment (choldgraf) [inline tutorials/02_intermediate/color_cycle.py]: space after `:,` (sorry I'm being persnickety about this)

File: tutorials/02_intermediate/color_cycle.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+# default prop_cycle using `plt.rc`. We'll combine a color cycler and a
```
Comment (choldgraf) [inline tutorials/02_intermediate/color_cycle.py]: here could you try linking to rc with an rST `:ref:`? I think that'll make it link nicely in the docs

File: tutorials/02_intermediate/color_cycle.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+plt.rc('axes', prop_cycle=default_cycler)
+
+###############################################################################
+# Now we'll generate a figure with two axes, one on top of the other. On the
+# first axis, we'll plot with the default cycler.
+fig, (ax0, ax1) = plt.subplots(nrows=2)
+ax0.plot(yy)
+ax0.set_title('Set default color cycle to rgby')
+
+###############################################################################
```
Comment (choldgraf) [inline tutorials/02_intermediate/color_cycle.py]: We probably don't want the line of `######` here since we're plotting on the same figure

Comment (choldgraf) [inline tutorials/02_intermediate/color_cycle.py]: just to clarify, in general we shouldn't split a single figure between two `#######` blocks of code

**Comment by choldgraf** [comment]
Just had a closer look - it looks really nice. A few styling comments to take care of and then once the tests pass I'm +1. Lemme know when you take care of that @johnh2o2

**Comment by WeatherGod** [comment]
Ping! This definitely would be a nice improvement to have. Let us know if you are having difficulties with rebasing.

**Comment by johnh2o2** [comment]
Hi @choldgraf, sorry for the massive delay -- I think I've implemented all of your suggested changes, and I've gotten `python make.py html` compile without problems and the pages look ok on my end. Let me know if there are any more issues with this.

Thanks!

**Comment by choldgraf** [comment]
hey @johnh2o2 - can you rename the paths so that the `02_` is removed? In the latest version we've removed those numbers from the paths so this will throw an error. Other than that LGTM.

**Comment by choldgraf** [comment]
pinging @johnh2o2 :-)

**Comment by johnh2o2** [comment]
Hey @choldgraf , so I'm not sure that I've done things correctly with the rebasing....I apologize if I've caused more problems...

(I THINK) I've rebased with the upstream/master branch, but now running `python make.py --allowsphinxwarnings html` gives me a number of warnings about unreferenced documentation:

```
checking consistency... /Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/lines_bars_and_markers/scatter_profile.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/agg_buffer.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/agg_buffer_to_array.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/anchored_artists.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/anscombe.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/arctest.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/arrow_demo.rst: WARNING: document isn't included in any toctree
/Users/jah5/Documents/scipy2017/sprints/matplotlib/doc/gallery/pylab_examples/arrow_simple_demo.rst: WARNING: document isn't included in any toctree
...
```

Not sure if this is a result of failed rebasing or if these are acceptable warnings or what. Let me know what I should (or shouldn't) do on my end.

Thanks!

John

**Comment by choldgraf** [comment]
whoah! it looks like there are now 636 files changed so that is almost certainly not correct haha. To be honest I think your best bet is to copy the test from the example that you've added (AKA the changes that were part of this initial PR), and start from scratch. I know that's a pain but it sounds like something very strange happened with the rebase.

How did you do the rebase? The correct commands from your fork would have been:

```
git checkout master
git pull upstream master
git checkout doc-improvements
git rebase master
```

**Comment by choldgraf** [comment]
closed via #9408 ...thanks @johnh2o2 !!!

**Pull request closed without merging**